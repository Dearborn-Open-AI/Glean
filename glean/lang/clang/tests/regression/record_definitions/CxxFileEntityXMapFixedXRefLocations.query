query: codemarkup.cxx.CxxFileEntityXMapFixedXRefLocations _
transform: [gensort, []]
