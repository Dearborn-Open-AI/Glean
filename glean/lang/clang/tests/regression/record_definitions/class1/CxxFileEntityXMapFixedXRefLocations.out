[
  "@generated",
  {
    "key": {
      "trace": {
        "key": {
          "xmap": {
            "key": {
              "file": { "key": "test.cpp" },
              "fixed": [
                {
                  "target": {
                    "declaration": {
                      "record_": {
                        "key": {
                          "name": {
                            "key": {
                              "name": { "key": "U" },
                              "scope": { "global_": { } }
                            }
                          },
                          "kind": { "struct_": { } },
                          "source": {
                            "file": { "key": "test.cpp" },
                            "lineBegin": 10,
                            "columnBegin": 1,
                            "lineEnd": 14,
                            "columnEnd": 1
                          }
                        }
                      }
                    }
                  },
                  "ranges": [
                    { "offset": 262, "length": 1 },
                    { "offset": 7, "length": 1 },
                    { "offset": 19, "length": 1 },
                    { "offset": 61, "length": 1 }
                  ]
                },
                {
                  "target": {
                    "declaration": {
                      "record_": {
                        "key": {
                          "name": {
                            "key": {
                              "name": { "key": "A" },
                              "scope": { "global_": { } }
                            }
                          },
                          "kind": { "struct_": { } },
                          "source": {
                            "file": { "key": "test.cpp" },
                            "lineBegin": 16,
                            "columnBegin": 1,
                            "lineEnd": 18,
                            "columnEnd": 1
                          }
                        }
                      }
                    }
                  },
                  "ranges": [
                    { "offset": 319, "length": 1 },
                    { "offset": 24, "length": 1 }
                  ]
                }
              ],
              "variable": [ ]
            }
          },
          "externals": [ ]
        }
      },
      "xref": {
        "target": {
          "name": "A",
          "file": { "key": "test.cpp" },
          "location": {
            "range": {
              "file": { "key": "test.cpp" },
              "lineBegin": 16,
              "columnBegin": 1,
              "lineEnd": 18,
              "columnEnd": 1
            }
          },
          "span": { "start": 304, "length": 1 }
        },
        "source": { "span": { "start": 319, "length": 1 } }
      },
      "entity": {
        "decl": {
          "record_": {
            "key": {
              "name": {
                "key": { "name": { "key": "A" }, "scope": { "global_": { } } }
              },
              "kind": { "struct_": { } },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 16,
                "columnBegin": 1,
                "lineEnd": 18,
                "columnEnd": 1
              }
            }
          }
        }
      }
    }
  },
  {
    "key": {
      "trace": {
        "key": {
          "xmap": {
            "key": {
              "file": { "key": "test.cpp" },
              "fixed": [
                {
                  "target": {
                    "declaration": {
                      "record_": {
                        "key": {
                          "name": {
                            "key": {
                              "name": { "key": "U" },
                              "scope": { "global_": { } }
                            }
                          },
                          "kind": { "struct_": { } },
                          "source": {
                            "file": { "key": "test.cpp" },
                            "lineBegin": 10,
                            "columnBegin": 1,
                            "lineEnd": 14,
                            "columnEnd": 1
                          }
                        }
                      }
                    }
                  },
                  "ranges": [
                    { "offset": 262, "length": 1 },
                    { "offset": 7, "length": 1 },
                    { "offset": 19, "length": 1 },
                    { "offset": 61, "length": 1 }
                  ]
                },
                {
                  "target": {
                    "declaration": {
                      "record_": {
                        "key": {
                          "name": {
                            "key": {
                              "name": { "key": "A" },
                              "scope": { "global_": { } }
                            }
                          },
                          "kind": { "struct_": { } },
                          "source": {
                            "file": { "key": "test.cpp" },
                            "lineBegin": 16,
                            "columnBegin": 1,
                            "lineEnd": 18,
                            "columnEnd": 1
                          }
                        }
                      }
                    }
                  },
                  "ranges": [
                    { "offset": 319, "length": 1 },
                    { "offset": 24, "length": 1 }
                  ]
                }
              ],
              "variable": [ ]
            }
          },
          "externals": [ ]
        }
      },
      "xref": {
        "target": {
          "name": "A",
          "file": { "key": "test.cpp" },
          "location": {
            "range": {
              "file": { "key": "test.cpp" },
              "lineBegin": 16,
              "columnBegin": 1,
              "lineEnd": 18,
              "columnEnd": 1
            }
          },
          "span": { "start": 304, "length": 1 }
        },
        "source": { "span": { "start": 343, "length": 1 } }
      },
      "entity": {
        "decl": {
          "record_": {
            "key": {
              "name": {
                "key": { "name": { "key": "A" }, "scope": { "global_": { } } }
              },
              "kind": { "struct_": { } },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 16,
                "columnBegin": 1,
                "lineEnd": 18,
                "columnEnd": 1
              }
            }
          }
        }
      }
    }
  },
  {
    "key": {
      "trace": {
        "key": {
          "xmap": {
            "key": {
              "file": { "key": "test.cpp" },
              "fixed": [
                {
                  "target": {
                    "declaration": {
                      "record_": {
                        "key": {
                          "name": {
                            "key": {
                              "name": { "key": "U" },
                              "scope": { "global_": { } }
                            }
                          },
                          "kind": { "struct_": { } },
                          "source": {
                            "file": { "key": "test.cpp" },
                            "lineBegin": 10,
                            "columnBegin": 1,
                            "lineEnd": 14,
                            "columnEnd": 1
                          }
                        }
                      }
                    }
                  },
                  "ranges": [
                    { "offset": 262, "length": 1 },
                    { "offset": 7, "length": 1 },
                    { "offset": 19, "length": 1 },
                    { "offset": 61, "length": 1 }
                  ]
                },
                {
                  "target": {
                    "declaration": {
                      "record_": {
                        "key": {
                          "name": {
                            "key": {
                              "name": { "key": "A" },
                              "scope": { "global_": { } }
                            }
                          },
                          "kind": { "struct_": { } },
                          "source": {
                            "file": { "key": "test.cpp" },
                            "lineBegin": 16,
                            "columnBegin": 1,
                            "lineEnd": 18,
                            "columnEnd": 1
                          }
                        }
                      }
                    }
                  },
                  "ranges": [
                    { "offset": 319, "length": 1 },
                    { "offset": 24, "length": 1 }
                  ]
                }
              ],
              "variable": [ ]
            }
          },
          "externals": [ ]
        }
      },
      "xref": {
        "target": {
          "name": "U",
          "file": { "key": "test.cpp" },
          "location": {
            "range": {
              "file": { "key": "test.cpp" },
              "lineBegin": 10,
              "columnBegin": 1,
              "lineEnd": 14,
              "columnEnd": 1
            }
          },
          "span": { "start": 241, "length": 1 }
        },
        "source": { "span": { "start": 262, "length": 1 } }
      },
      "entity": {
        "decl": {
          "record_": {
            "key": {
              "name": {
                "key": { "name": { "key": "U" }, "scope": { "global_": { } } }
              },
              "kind": { "struct_": { } },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 10,
                "columnBegin": 1,
                "lineEnd": 14,
                "columnEnd": 1
              }
            }
          }
        }
      }
    }
  },
  {
    "key": {
      "trace": {
        "key": {
          "xmap": {
            "key": {
              "file": { "key": "test.cpp" },
              "fixed": [
                {
                  "target": {
                    "declaration": {
                      "record_": {
                        "key": {
                          "name": {
                            "key": {
                              "name": { "key": "U" },
                              "scope": { "global_": { } }
                            }
                          },
                          "kind": { "struct_": { } },
                          "source": {
                            "file": { "key": "test.cpp" },
                            "lineBegin": 10,
                            "columnBegin": 1,
                            "lineEnd": 14,
                            "columnEnd": 1
                          }
                        }
                      }
                    }
                  },
                  "ranges": [
                    { "offset": 262, "length": 1 },
                    { "offset": 7, "length": 1 },
                    { "offset": 19, "length": 1 },
                    { "offset": 61, "length": 1 }
                  ]
                },
                {
                  "target": {
                    "declaration": {
                      "record_": {
                        "key": {
                          "name": {
                            "key": {
                              "name": { "key": "A" },
                              "scope": { "global_": { } }
                            }
                          },
                          "kind": { "struct_": { } },
                          "source": {
                            "file": { "key": "test.cpp" },
                            "lineBegin": 16,
                            "columnBegin": 1,
                            "lineEnd": 18,
                            "columnEnd": 1
                          }
                        }
                      }
                    }
                  },
                  "ranges": [
                    { "offset": 319, "length": 1 },
                    { "offset": 24, "length": 1 }
                  ]
                }
              ],
              "variable": [ ]
            }
          },
          "externals": [ ]
        }
      },
      "xref": {
        "target": {
          "name": "U",
          "file": { "key": "test.cpp" },
          "location": {
            "range": {
              "file": { "key": "test.cpp" },
              "lineBegin": 10,
              "columnBegin": 1,
              "lineEnd": 14,
              "columnEnd": 1
            }
          },
          "span": { "start": 241, "length": 1 }
        },
        "source": { "span": { "start": 269, "length": 1 } }
      },
      "entity": {
        "decl": {
          "record_": {
            "key": {
              "name": {
                "key": { "name": { "key": "U" }, "scope": { "global_": { } } }
              },
              "kind": { "struct_": { } },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 10,
                "columnBegin": 1,
                "lineEnd": 14,
                "columnEnd": 1
              }
            }
          }
        }
      }
    }
  },
  {
    "key": {
      "trace": {
        "key": {
          "xmap": {
            "key": {
              "file": { "key": "test.cpp" },
              "fixed": [
                {
                  "target": {
                    "declaration": {
                      "record_": {
                        "key": {
                          "name": {
                            "key": {
                              "name": { "key": "U" },
                              "scope": { "global_": { } }
                            }
                          },
                          "kind": { "struct_": { } },
                          "source": {
                            "file": { "key": "test.cpp" },
                            "lineBegin": 10,
                            "columnBegin": 1,
                            "lineEnd": 14,
                            "columnEnd": 1
                          }
                        }
                      }
                    }
                  },
                  "ranges": [
                    { "offset": 262, "length": 1 },
                    { "offset": 7, "length": 1 },
                    { "offset": 19, "length": 1 },
                    { "offset": 61, "length": 1 }
                  ]
                },
                {
                  "target": {
                    "declaration": {
                      "record_": {
                        "key": {
                          "name": {
                            "key": {
                              "name": { "key": "A" },
                              "scope": { "global_": { } }
                            }
                          },
                          "kind": { "struct_": { } },
                          "source": {
                            "file": { "key": "test.cpp" },
                            "lineBegin": 16,
                            "columnBegin": 1,
                            "lineEnd": 18,
                            "columnEnd": 1
                          }
                        }
                      }
                    }
                  },
                  "ranges": [
                    { "offset": 319, "length": 1 },
                    { "offset": 24, "length": 1 }
                  ]
                }
              ],
              "variable": [ ]
            }
          },
          "externals": [ ]
        }
      },
      "xref": {
        "target": {
          "name": "U",
          "file": { "key": "test.cpp" },
          "location": {
            "range": {
              "file": { "key": "test.cpp" },
              "lineBegin": 10,
              "columnBegin": 1,
              "lineEnd": 14,
              "columnEnd": 1
            }
          },
          "span": { "start": 241, "length": 1 }
        },
        "source": { "span": { "start": 288, "length": 1 } }
      },
      "entity": {
        "decl": {
          "record_": {
            "key": {
              "name": {
                "key": { "name": { "key": "U" }, "scope": { "global_": { } } }
              },
              "kind": { "struct_": { } },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 10,
                "columnBegin": 1,
                "lineEnd": 14,
                "columnEnd": 1
              }
            }
          }
        }
      }
    }
  },
  {
    "key": {
      "trace": {
        "key": {
          "xmap": {
            "key": {
              "file": { "key": "test.cpp" },
              "fixed": [
                {
                  "target": {
                    "declaration": {
                      "record_": {
                        "key": {
                          "name": {
                            "key": {
                              "name": { "key": "U" },
                              "scope": { "global_": { } }
                            }
                          },
                          "kind": { "struct_": { } },
                          "source": {
                            "file": { "key": "test.cpp" },
                            "lineBegin": 10,
                            "columnBegin": 1,
                            "lineEnd": 14,
                            "columnEnd": 1
                          }
                        }
                      }
                    }
                  },
                  "ranges": [
                    { "offset": 262, "length": 1 },
                    { "offset": 7, "length": 1 },
                    { "offset": 19, "length": 1 },
                    { "offset": 61, "length": 1 }
                  ]
                },
                {
                  "target": {
                    "declaration": {
                      "record_": {
                        "key": {
                          "name": {
                            "key": {
                              "name": { "key": "A" },
                              "scope": { "global_": { } }
                            }
                          },
                          "kind": { "struct_": { } },
                          "source": {
                            "file": { "key": "test.cpp" },
                            "lineBegin": 16,
                            "columnBegin": 1,
                            "lineEnd": 18,
                            "columnEnd": 1
                          }
                        }
                      }
                    }
                  },
                  "ranges": [
                    { "offset": 319, "length": 1 },
                    { "offset": 24, "length": 1 }
                  ]
                }
              ],
              "variable": [ ]
            }
          },
          "externals": [ ]
        }
      },
      "xref": {
        "target": {
          "name": "U",
          "file": { "key": "test.cpp" },
          "location": {
            "range": {
              "file": { "key": "test.cpp" },
              "lineBegin": 10,
              "columnBegin": 1,
              "lineEnd": 14,
              "columnEnd": 1
            }
          },
          "span": { "start": 241, "length": 1 }
        },
        "source": { "span": { "start": 349, "length": 1 } }
      },
      "entity": {
        "decl": {
          "record_": {
            "key": {
              "name": {
                "key": { "name": { "key": "U" }, "scope": { "global_": { } } }
              },
              "kind": { "struct_": { } },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 10,
                "columnBegin": 1,
                "lineEnd": 14,
                "columnEnd": 1
              }
            }
          }
        }
      }
    }
  }
]